stringlength.md

```diff
@@ -0,0 +1,112 @@
+# Proposal to add Suggested Lengths for String Signals in the VSS
+
+This document covers a concept proposal to add to the VSS .vspec files an additional keyword 
+(`stringsize:` or similar) to assist the code generators (in vss-tools) in setting an appropriate
+default size for strings.
+
+This new keyword would be added to the .vspec files for all `datatype: string` elements (approx. 85 places),

```

Comment:
I think this is a useful keyword, but only if it is optional: I.e. I would set a stringsize in standard catalogue only, when it makes semantic sense (e.g. DTCs, where we know they look like X0000).
In case of allowed values the stringlenth is implicitly clear

There are however places where you want to model an unbounded string (Of course your implementation/generator might need to enforce bounds but that is either deployment specific configuration, or implicit in the  tooling). One example: If you had a model defining a Drivers's name. Obviously you implementation might not accept a 1MiB string, but from a data modelling perpsective it does not make sense to limit to something on the base VSS level.

tldr.: Like the keyword, if it is optional (Tooling/exporters that requires it might just substitute a default value for unbounded strings, and recommend you adding a  specific overlay)